seed/seed20.go

```diff
@@ -580,9 +580,15 @@ func (s *seed20) doLoadMeta(handler SnapHandler, tm timings.Measurer) error {
 	}
 	stopCh := make(chan struct{})
 	outcomesCh := make(chan error, njobs)
+	var wg sync.WaitGroup
+	wg.Add(njobs)
 	for j := 1; j <= njobs; j++ {
 		jtm := tm.StartSpan(fmt.Sprintf("do-load-meta[%d]", j), fmt.Sprintf("snap metadata loading job #%d", j))
 		go func() {
+			// defers are LIFO, make sure that time snap is stopped
+			// before we let the parent know that the goroutine is
+			// done
+			defer wg.Done()
 			defer jtm.Stop()
```

Comment:
How about this:

```go
var retErr error
defer func() {
    outcomesCh <- retErr
}()
defer jtm.Stop()
...
        if err == errSkipped {
            continue
        }
        retErr = err
        return
...
```

This way, we use defer to ensure `jtm.Stop()` is always closed, no matter which return is taken, and the defer stack ensures `jtm.Stop()` is called before sending over the channel.

Reply:
and pushed